modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -880,4 +880,37 @@ public void setPerformBeanValidation(boolean performBeanValidation) {
     public void setUseOptional(boolean useOptional) {
         this.useOptional = useOptional;
     }
-}
+
+    @Override
+    public void postProcessParameter(CodegenParameter p) {
+        // we use a custom version of this function to remove the l, d, and f suffixes from Long/Double/Float
+        // defaultValues
+        // remove the l because our users will use Long.parseLong(String defaultValue)
+        // remove the d because our users will use Double.parseDouble(String defaultValue)
+        // remove the f because our users will use Float.parseFloat(String defaultValue)
+        // NOTE: for CodegenParameters we DO need these suffixes because those defaultValues are used as java value
+        // literals assigned to Long/Double/Float
+        if (p.defaultValue == null) {
+            return;
+        }
+        Boolean fixLong = (p.isLong && "l".equals(p.defaultValue.substring(p.defaultValue.length()-1)));
+        Boolean fixDouble = (p.isDouble && "d".equals(p.defaultValue.substring(p.defaultValue.length()-1)));
+        Boolean fixFloat = (p.isFloat && "f".equals(p.defaultValue.substring(p.defaultValue.length()-1)));
+        if (fixLong || fixDouble || fixFloat) {
+            p.defaultValue = p.defaultValue.substring(0, p.defaultValue.length()-1);
+        }
+    }
+
+    @Override
+    public void addImportsToOneOfInterface(List<Map<String, String>> imports) {
+        for (String i : Arrays.asList("JsonSubTypes", "JsonTypeInfo")) {
```

Comment:
There should be a check for JACKSON library. Otherwise, code with GSON will be not compilable.
See PR https://github.com/OpenAPITools/openapi-generator/pull/10286

Reply:
Thanks for the review. Can you please submit a PR against this branch with the suggested fix?

Reply:
@wing328 I wonder if you could check the PR https://github.com/OpenAPITools/openapi-generator/pull/10286 .
It fixes inline OneOf Model generating and One of without discriminator cases.
If it works - I will recreate PR

Reply:
@wing328 I created PR against jburgess-spring_fix_5381_jb https://github.com/OpenAPITools/openapi-generator/pull/10433/ and resolved conflicts